draft-ietf-jsonpath-base.md

```diff
@@ -879,26 +753,80 @@ JSON:
 
 Queries:
 
+The following examples show the `wildcard` selector in use by a child __APPENDER__.
+
 | Query | Result | Result Paths | Comment |
 | :---: | ------ | :----------: | ------- |
+| `$[*]`   | `{"j": 1, "k": 2}` <br> `[5, 3]` | `$['o']` <br> `$['a']` | Object values      |
 | `$.o[*]` | `1` <br> `2` | `$['o']['j']` <br> `$['o']['k']` | Object values      |
 | `$.o[*]` | `2` <br> `1` | `$['o']['k']` <br> `$['o']['j']` | Alternative result |
 | `$.a[*]` | `5` <br> `3` | `$['a'][0]` <br> `$['a'][1]`     | Array members      |
-{: title="Index wildcard selector examples"}
+{: title="Wildcard selector examples"}
+
+### Index selector {#index-selector}
+
+#### Syntax {#syntax-index}
+{: unnumbered}
+
+An index selector `<index>` matches at most one array element value.
+
+~~~~ abnf
+index-selector  = int                             ; decimal integer
+
+int             = ["-"] ( "0" / (DIGIT1 *DIGIT) ) ; -  optional
+DIGIT1          = %x31-39                         ; 1-9 non-zero digit
+~~~~
+
+Applying the numerical `index-selector` is required to select the corresponding
+element. JSONPath allows it to be negative (see {{index-semantics}}).
+
+Notes:
+1. An `index-selector` is an integer (in base 10, as in JSON numbers).
+2. As in JSON numbers, the syntax does not allow octal-like integers with leading zeros such as `01` or `-01`.
 
-### Array Slice Selector {#slice}
+#### Semantics {#index-semantics}
+{: unnumbered}
+
+The `index-selector` applied to an array selects an array element using a zero-based index.
```

Comment:
Should we say what happens when it is not applied to an array?
(Nothing is selected.)

Reply:
Er, that paragraph ends with:

> Nothing is selected from a value that is not an array.

Reply:
Sounds good :-)